lib/services/local/dynamic_access.go

```diff
@@ -101,6 +112,11 @@ func (s *DynamicAccessService) GetAccessRequests(ctx context.Context, filter ser
 	if filter.ID != "" {
 		req, err := s.GetAccessRequest(ctx, filter.ID)
 		if err != nil {
+			if trace.IsNotFound(err) {
+				// Caller is going to need to check the length of the returned slice anyway,
```

Comment:
I did not understand this comment. If something is not found, error should be returned, there should not be a scenario when we return nil, nil so to speak.